src/lib.rs

```diff
@@ -213,7 +216,7 @@ impl<'a> WriteBase32 for Bech32Writer<'a> {
 
 impl<'a> Drop for Bech32Writer<'a> {
     fn drop(&mut self) {
-        self.inner_finalize()
+        self.write_checksum()
             .expect("Unhandled error writing the checksum on drop.")
```

Comment:
FTR `std` ignores errors instead of panicking in drop and it is considered the right thing to do because panicking in drop is a footgun.

Reply:
Lol, oops! I tried to look for double-panics but got a bit confused and somehow missed this completely obvious one.

But I think we should remove this `Drop` business entirely so it's fine for now. I've never heard of a wrapper struct that takes a `&mut` ref and then has finalization logic on `Drop`...I suspect this was an API experiment that never failed enough to be removed.

Reply:
IDK, it's not too different from `BufWriter` behavior. But I guess compiler warning would be better than drop. I was thinking of abusing `must_use` but then we couldn't implement any methods taking `&mut self` and it'd be fragile anyway.